### PR TITLE
Migrate from CRA to Vite.js

### LINKED ISSUE #81: Switch app from CRA to Vite

### PR DESCRIPTION
**Pull request by ashwanijha1996**
## Fixes Issue

closes #81 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by ashwanijha1996** [comment]
Hi @furkanmutlu I have successfully migrated this webapp from CRA to Vite.js. Please have a review and suggest change if any. Thank you.

**Comment by furkanmutlu** [comment]
Please squash / out all your commits into 1. Seems like you have 12 commits,
1- it is confusing
2- in case of an issue, it would be harder to rollback.
3- It is a good practice to have minimum amount of commits

File: frontend/.env.example [hunk 1]
```diff
@@ -1 +1 @@
-REACT_APP_API_URL=http://localhost:5000
\ No newline at end of file
+VITE_APP_API_URL=http://localhost:5000
```
Comment (furkanmutlu) [inline frontend/.env.example:1]: Please add extra line at the end

File: frontend/tsconfig.json
Comment (furkanmutlu) [inline frontend/tsconfig.json:1]: Why adding TypeScript?

File: frontend/package.json [hunk 2]
```diff
@@ -20,16 +21,14 @@
     "react-dom": "^16.14.0",
     "react-lazy-load-image-component": "^1.5.6",
     "react-router-dom": "^6.6.1",
-    "react-scripts": "3.4.3",
-    "sass": "^1.49.9"
+    "sass": "^1.49.9",
+    "vite": "^5.0.10",
+    "vite-tsconfig-paths": "^4.2.2"
```
Comment (furkanmutlu) [inline frontend/package.json:26]: there is no need to Typescript

File: frontend/vite-env.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+/// <reference types="vite/client" />
```
Comment (furkanmutlu) [inline frontend/vite-env.d.ts:1]: What is this for?

File: frontend/src/utils/getImages.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { LIMIT } from "../static/constants";
 
-const axios = require("axios");
+import axios from "axios";
 const hostName = `${
-  process?.env?.REACT_APP_API_URL || "http://localhost:5000"
+  import.meta?.env?.VITE_APP_API_URL || "http://localhost:5000"
```
Comment (furkanmutlu) [inline frontend/src/utils/getImages.js:5]: hmm this changes things will also break the production. I will need to adjust the environment variables.

**Review by furkanmutlu — COMMENTED** [review]
I think this task a bit more complicated, then I estimated. Since it would also require changes on environment variables in production can I take this issue?

**Pull request closed without merging**